Cheatsheet.md

```diff
@@ -0,0 +1,504 @@
+
+**Syntax:**
+ * _array.join(separator)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| separator	| Optional.   | 
+
+**Return value:**
+String, representing the array values, separated by the specified separator.
+
+**Example:**
+```
+const elements = [\'geeksforgeeks\', \'gfg\']; console.log(elements.join()); 
+console.log(elements.join(\'.\')); 
+console.log(elements.join(\'-\')); 
+```
+*Output:*
+
+    > \"geeksforgeeks, gfg\"
+    > \"geeksforgeeks.gfg\"
+    > \"geeksforgeeks-gfg\"
+
+### [8. push](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/push)
+
+**Description:**
+Adds new items to the end of an array, and returns the new length.
+
+**Syntax:**
+ * _array.push(item1, item2, ..., itemX)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| item1, item2, ..., itemX | Required. The item(s) to add to the array |
+
+**Return value:**
+New length of the array.
+
+**Example:**
+```
+let arr = [34, 234, 567, 4];
+console.log(arr.push(23,45,56));
+console.log(arr);
+```
+*Output:*
+
+    7
+    34,234,567,4,23,45,56
+
+### [9. find](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/find)
+
+**Description:**
+Returns the value of the first element in an array that pass a test (provided as a function).
+
+**Syntax:**
+ * _array.find(function(currentValue, index, arr),thisValue)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| function(currentValue, index, arr) |	Required. A function to be run for each element in the array.|
+| thisValue	| Optional. |
+
+**Return value:**
+ The array element value if any of the elements in the array pass the test, otherwise it returns 'undefined'.
+
+**Example:**
+ ```  
+ let array = [2, 7, 8, 9]; 
+ const found = array.find(element=>element > 4); 
+ // Printing desired values. 
+ console.log(found); 
+```
+*Output:*
+    
+    > 7
+
+### [10. indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
+
+**Description:**
+returns the position of the first occurrence of a specified value in a string.
+
+**Syntax:**
+ * _string.indexOf(searchvalue, start)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| searchvalue | Required. The string to search for |
+| start	      | Optional. <br>Default 0. At which position to start the search.|
+
+**Return value:**
+Number, representing the position where the specified searchvalue occurs for the first time, or -1 if it never occurs.
+
+**Example:**
+```
+console.log('Departed Train'.indexOf('Train'));
+```
+*Output:*
+    
+    9
+    
+## Commonly used functions on [string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) 
+
+### [1. substring](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/substring)
+
+**Description:**
+Extracts the characters in a string between "start" and "end", not including "end" itself.

```

Comment:
Add a note with the similarities/differences between this and array.slice since beginners get confused on them